libraries/chain/include/eosio/chain/backing_store/chain_kv_payer.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <eosio/chain/types.hpp>
+
+namespace eosio { namespace chain { namespace backing_store {
+   static constexpr auto payer_in_value_size = sizeof(account_name);
+
+   inline static uint32_t actual_value_size(const uint32_t raw_value_size) {
+      EOS_ASSERT(raw_value_size >= payer_in_value_size, kv_rocksdb_bad_value_size_exception , "The size of value returned from RocksDB is less than payer's size");
+      return (raw_value_size - payer_in_value_size);
+   }
+
+   inline static account_name get_payer(const char* data) {
+      account_name payer;
+      memcpy(&payer, data, payer_in_value_size);
+      return payer;
+   }
+
+   inline static const char* actual_value_start(const char* data) {
+      return data + payer_in_value_size;
+   }
+
+}}} // ns eosio::chain::backing_store
```

Comment:
Very good abstraction. After this PR is merged, I can move the rest of payer related functions in kv_rocksdb_context.h here and both of us use this same header file.

Reply:
You have already done that below! Thanks!

Reply:
After this goes in, I'd like to remove the namespace methods and have everything use the payer_payload directly.  That will mean that in some scenarios there will be an unneeded conversion from bytes to a payer, but I think that is in the noise and made up for in ensuring that all values are verified (like verifying there is a payer's worth of bytes before retrieving a payer)